analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/ProgramStateTest.CaptureOperation.cs

```diff
@@ -106,4 +106,23 @@ public void ResolveCapture_OtherOperation_ReturnsArgument()
         var sut = ProgramState.Empty.SetCapture(capture.Id, capture.Value);
         sut.ResolveCapture(capture.Value).Should().Be(capture.Value);   // Any other operation
     }
+
+    [TestMethod]
+    public void ResolveCaptureAndUnwrapConversion_CaptureReferenceInConversion_ReturnsCapturedOperation()
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("_ = (condition ? a : null) as string;", "object a, bool condition");
+        var capture = IFlowCaptureOperationWrapper.FromOperation(cfg.Blocks[2].Operations[0]);
+        var conversion = IConversionOperationWrapper.FromOperation(cfg.Blocks[4].Operations[0].ChildOperations.First().ChildOperations.Skip(1).First());
```

Comment:
This can be also done in this way:
```
var conversion = cfg.Blocks[4].Operations.ToExecutionOrder().Select(x => x.Instance).OfType<IConversionOperation>().Single();
```
Both are big ugly in different ways